lib/utils/changeUrlQueryString.ts

```diff
@@ -2,5 +2,10 @@ import { NextRouter } from "next/router";
 import { ParsedUrlQueryInput } from "querystring";
 
 export const changeUrlQueryString = (query: ParsedUrlQueryInput, router: NextRouter) => {
-  router.replace({ query: query }, undefined, { scroll: false, shallow: true });
+  const { envId, ...restQuery } = query; // get rid of envId
+  const asPath = {
```

Comment:
this comment doesn't say to me much why are getting rid of envid? better comment would explain why do we need asPath without the envid? is it because it is handled component and won't by the middleware?

Reply:
If I recall, the idea with products was that they'll be filtered client-side, unlike articles that are pulled from API. that was in pre-middleware times though, so I'm open to suggestions. In any case, the spread operator is used to get rid of `envId` in construction of the subsequent `asPath`, used in router.replace to ensure the envId is not present in the address bar.